pandas/io/parsers.py

```diff
@@ -3098,15 +3110,17 @@ def _clean_na_values(na_values, keep_default_na=True):
         na_fvalues = set()
     elif isinstance(na_values, dict):
         na_values = na_values.copy()  # Prevent aliasing.
-        if keep_default_na:
-            for k, v in compat.iteritems(na_values):
```

Comment:
might be nice to have a comment or 2 to explain the logic here for future readers

Reply:
Makes sense.  Done.